packages/react-search-ui-views/src/__tests__/view-helpers/formatResult.test.ts

```diff
@@ -0,0 +1,132 @@
+import { formatResult } from "../../view-helpers";
+
+describe("formatResult", () => {
+  it("formats strings correctly", () => {
+    expect(
+      formatResult({
+        stringField: { raw: "stringValue" }
+      })
+    ).toEqual({
+      stringField: "stringValue"
+    });
+  });
+
+  it("formats numbers correctly", () => {
+    expect(
+      formatResult({
+        numberField: { raw: 5 }
+      })
+    ).toEqual({
+      numberField: "5"
+    });
+  });
+
+  it("formats dates correctly", () => {
+    expect(
+      formatResult({
+        dateField: { raw: "1919-11-19T06:00:00+00:00" }
+      })
+    ).toEqual({
+      dateField: "1919-11-19T06:00:00+00:00"
+    });
+  });
+
+  it("formats string locations correctly", () => {
+    expect(
+      formatResult({
+        locationStringField: { raw: "41.12,-71.34" }
+      })
+    ).toEqual({
+      locationStringField: "41.12,-71.34"
+    });
+  });
+
+  it("formats array locations correctly", () => {
+    expect(
+      formatResult({
+        locationArrayField: { raw: [-71.34, 41.12] }
+      })
+    ).toEqual({
+      locationArrayField: "-71.34,41.12"
+    });
+  });
+
+  it("filters out arbitrary values", () => {
+    expect(
+      formatResult({
+        stringField: { raw: "stringValue" },
+        arbitraryField: "9000"
+      })
+    ).toEqual({
+      stringField: "stringValue"
+    });
+  });
+
+  it("filters out _meta field", () => {
+    expect(
+      formatResult({
+        stringField: { raw: "stringValue" },
+        _meta: {
+          id: "123",
+          engine: "national-parks",
+          score: 1
+        }
+      })
+    ).toEqual({
+      stringField: "stringValue"
+    });
+  });
+
+  it("html escapes raw values", () => {
```

Comment:
To be on the safe side, would it make sense to add a few tests here for some different variations? Like for nested fields, etc. where we go down different logic paths that execute the escape logic in different ways.